qualtrics_link/views.py

```diff
@@ -189,35 +189,20 @@ def internal(request):
         user_can_access = True
     
     if user_can_access:
-        # If they are allowed to use Qualtrics, check to see if the user has accepted the terms of service
-        user_acceptance = None
-        try:
-            user_acceptance = Acceptance.objects.get(user_id=huid)
-        except Acceptance.DoesNotExist:
-            logger.info('User %s has not  accepted term of service ', huid)
-        except Exception as e:
-            logger.error('Exception in checking for user acceptance, '
-                     'user_id:%s', huid, e)
-            return render(request, 'qualtrics_link/error.html')
 
         # Set the initial values of the Qualtrics Admin form
         qualtrics_user_update_form.initial['division'] = person_details.division
         qualtrics_user_update_form.initial['role'] = person_details.role
 
         qualtrics_user_in_db = False
         try:
-            qu = QualtricsUser.objects.get(univ_id=person_details.id)
-            qualtrics_user_update_form.initial['manually_updated'] = qu.manually_updated
-            qualtrics_user_in_db = True
+            qu = QualtricsUser.objects.filter(univ_id=person_details.id).first()
+            if qu:
+                qualtrics_user_update_form.initial['manually_updated'] = qu.manually_updated
+                qualtrics_user_in_db = True
         except QualtricsUser.DoesNotExist:
             qualtrics_user_update_form.initial['manually_updated'] = False
```

Comment:
This exception will no longer be hit if you are using filter over get. I would add in a else condition to the "if qu" to set the manually_updated initial value.

Reply:
Thanks. Removed the Exception and handled setting the manually_updated default value outside the block